### PR TITLE
#779: Unit tests for NetworkConfiguration class from core.net

### PR DESCRIPTION
**Pull request by djurod**
- Prepared unit tests for class NetworkConfiguration from package
  org.eclipse.kura.core.net
- Fixed an issue in isValid() and addWifiConfigIP4Properties() methods
  in NetworkConfiguration class
- Removed "m_" prefix from a method parameter name in WifiConfig class
- Implemented hashCode() and equals() methods in Password class

Signed-off-by: Djuro Drljaca <djuro.drljaca@comtrade.com>

### CODE REVIEW

**Comment by djurod** [comment]
As requested by @MMaiero the hashCode() and equals() methods in Password class were removed from the pull request.

File: kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java [hunk 1]
```diff
@@ -424,9 +422,9 @@ public boolean isValid() throws KuraException {
             }
 
             NetInterfaceType type = netInterfaceConfig.getType();
-            if (type != NetInterfaceType.ETHERNET && type != NetInterfaceType.WIFI
+            if (type != NetInterfaceType.ETHERNET && type != NetInterfaceType.WIFI && type != NetInterfaceType.MODEM
                     && type != NetInterfaceType.LOOPBACK) {
-                s_logger.error("Type must be ETHERNET, WIFI, or LOOPBACK - type is " + type);
+                s_logger.error("Type must be ETHERNET, WIFI, MODEM, or LOOPBACK - type is " + type);
```
Comment (cdealti) [inline kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java:427]: I've never seen this error message. So either there are no modem configurations, or the method is never called.

Comment (djurod) [reply kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java:427]: As far as I understand this part of code you usually shouldn't see this. This is called only if you prepare a NetInterfaceAddressConfig for the unexpected type which I would assume doesn't happen very often :)

Comment (cdealti) [reply kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java:427]: @ibinshtok do you know why originally didn't not check if the interface type was not a modem?

Comment (ibinshtok) [reply kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java:427]: @cdealti I don't know, and it looks to me that this method is never called.

**Comment by cdealti** [comment]
@djurod this has been manually merged

**Pull request closed without merging**